### PR TITLE
[Github Action] cli-test: improve naming summary

### PR DESCRIPTION
**Pull request by savil**
## Summary

**Motivation**
The name summary for `cli-tests` in the Github PR page is:
```
test (false, ubuntu-latest, true, 2.12.0)
```
This isn't very helpful.

This PR:
- removes the `booleans` and changes their true/false states to explicit names
- renames `devbox-json-tests` to `project-tests` for brevity.

## How was it tested?

will observe the testscripts pass...

Now, it looks like:
```
cli-tests / test (not-main, ubuntu-latest, project-tests, 2.12.0)
```
which is much more understandable.

### CODE REVIEW

**Review by LucilleH — APPROVED** [review]

File: .github/workflows/cli-tests.yaml [hunk 1]
```diff
@@ -76,24 +74,25 @@ jobs:
...
         nix-version: ["2.12.0", "2.17.0"]
         exclude:
-          - is-main: false
+          - is-main: "not-main"
             os: "${{ inputs.run-mac-tests && 'dummy' || 'macos-latest' }}"
-          - is-main: true
-            run-devbox-json-tests: false
-          - run-devbox-json-tests: true
+          - is-main: "is-main"
+            run-project-tests: "project-tests"
```
Comment (ipince) [inline .github/workflows/cli-tests.yaml:92]: shouldn't this be `project-tests-off`?

File: .github/workflows/cli-tests.yaml [hunk 2]
```diff
@@ -131,8 +130,8 @@ jobs:
         env:
           # For devbox.json tests, we default to non-debug mode since the debug output is less useful than for unit testscripts.
           # But we allow overriding via inputs.example-debug
-          DEVBOX_DEBUG: ${{ (!matrix.run-devbox-json-tests || inputs.example-debug) && '1' || '0' }}
-          DEVBOX_RUN_DEVBOX_JSON_TESTS: ${{ matrix.run-devbox-json-tests }}
+          DEVBOX_DEBUG: ${{ (matrix.run-project-tests == 'not-project-tests' || inputs.example-debug) && '1' || '0' }}
```
Comment (ipince) [inline .github/workflows/cli-tests.yaml:133]: shouldn't this be `project-tests-off`?

**Review by ipince — APPROVED** [review]
Approving to unblock, but see comments

File: .github/workflows/cli-tests.yaml [hunk 3]
```diff
@@ -76,24 +74,25 @@ jobs:
...
-        run-devbox-json-tests: [true, false]
+        run-project-tests: ["project-tests", "project-tests-off"]
         # Run tests on:
         # 1. the oldest supported nix version (which is 2.9.0? But determinate-systems installer has 2.12.0)
         # 2. nix version 2.17.0 which introduces a new code path that minimizes nixpkgs downloads.
         # 3. latest nix version (currently, that is 2.17.0, so omitted)
         nix-version: ["2.12.0", "2.17.0"]
         exclude:
-          - is-main: false
+          - is-main: "not-main"
```
Comment (mikeland73) [inline .github/workflows/cli-tests.yaml:89]: I agree that improving the labels is good, but lines like this make my head spin :(

**Review by mikeland73 — COMMENTED** [review]
I totally agree the labels suck, but I really have trouble following our rules (Many of which I wrote myself - lol)

Alternative:

* Create two different workflows that both call a job in a shared workflow.
* The shared workflow takes inputs that decide what to run (what OS, and what scope of tests).


So the jobs in non-main can be named:
* Basic tests on mac
* All tests on linux

On main:
* All tests on (mac|linux)

(under the hood, each of these is just calling some shared jobs in another workflow which is never triggered directly).

This should really simplify names and also remove much of the exclude logic which is hard to grok.

File: .github/workflows/cli-tests.yaml (hunk 2 above)
Comment (savil) [reply .github/workflows/cli-tests.yaml]: good catch!

**Review by savil — COMMENTED** [review]
@mikeland73 some refactoring like that seems called for. I'll circle back to doing that in a bit.

Will land this for now, so at least we get rid of boolean labels.

**Pull request merged**